packages/thrift-client/src/main/connections/HttpConnection.ts

```diff
@@ -25,58 +24,56 @@ export const DEFAULT_PATH: string = '/thrift'
 
 export type HttpProtocol = 'http' | 'https'
 
-export type RequestInstance = RequestAPI<
-    Request,
-    RequestOptions,
-    RequiredUriUrl
->
-
 function shouldRetry(
-    response: RequestResponse,
+    response: Response,
     retry: boolean,
     withEndpointPerMethod: boolean,
 ): boolean {
-    return (
-        withEndpointPerMethod &&
-        response !== undefined &&
-        response !== null &&
-        response.statusCode !== undefined &&
-        response.statusCode === 404 &&
-        retry === false
-    )
+    return withEndpointPerMethod && response.status === 404 && retry === false
```

Comment:
will `response` always be defined now? seems like old code didn't think so.

Reply:
Yes, it will be - not sure why they had those checks in there (perhaps the `request` library had it possibly `undefined`), but we'll always have it if we're calling these functions.